droidlet/interpreter/robot/tasks.py

```diff
@@ -350,3 +354,100 @@ def step(self):
             self.agent.mover.explore()
         else:
             self.finished = self.agent.mover.bot_step()
+        
+
+class CuriousExplore(Task):
+    """use slam to explore environemt, but also examine detections"""
+
+    def __init__(self, agent, task_data):
+        super().__init__(agent)
+        self.steps = ["not_started"] * 2
+        self.init_curious_logger()
+        self.agent = agent
+        TaskNode(agent.memory, self.memid).update_task(task=self)
+
+    def init_curious_logger(self):
+        self.logger = logging.getLogger('curious')
+        self.logger.setLevel(logging.INFO)
+        fh = logging.FileHandler('curious_explore.log')
+        fh.setLevel(logging.INFO)
+        ch = logging.StreamHandler()
+        ch.setLevel(logging.INFO)
+        formatter = logging.Formatter('%(filename)s:%(lineno)s - %(funcName)s(): %(message)s')
+        fh.setFormatter(formatter)
+        self.logger.addHandler(fh)
+        self.logger.addHandler(ch)
+
+    @Task.step_wrapper
+    def step(self):
+        self.interrupted = False
+        self.finished = False
+       
+        # execute a examine maneuver
+        if self.steps[0] == "not_started":
+            # Get a list of current detections
+            objects = DetectedObjectNode.get_all(self.agent.memory)
+            pos = self.agent.mover.get_base_pos_in_canonical_coords()
+            # pick all from unexamined, in-sight object
+            def pick_random_in_sight(objects, base_pos):
+                for x in objects:
+                    if ExaminedMap.can_examine(x):
```

Comment:
see above, would really rather this object is stored in memory, even in a shallow way that is not linked to rest of db

Reply:
(but make a task to integrate properly)

Reply:
also is this a global?  I don't understand where it lives